Add hints towards flash protection to all items which provide it

Helix-ctrl opened the pull request:
## About the PR
All items which provide protection from flashes will say so when examined
## Why / Balance
I spoke to a few new players none of which knew that welding masks even provided flash protection, even after a week of playtime.

## Technical details

Added subscriberlocalevent to FlashSystem.cs displaying examine text referencing flash-immunity-component.ftl 
Created flash-immunity-component.ftl

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
![image](https://github.com/user-attachments/assets/87a2228a-8c8c-45ef-bdab-b4720ed64424)


## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->

🆑 
Items which provide flash protection will now display it upon being examined

EmoGarbage404 commented:
Instead of being a description I feel like this should just be automatic examine text that gets applied.

Helix-ctrl commented:
> Instead of being a description I feel like this should just be automatic examine text that gets applied.

I'm not sure how i would do that

EmoGarbage404 commented:
you would need to add a subscription for `ExaminedEvent` on `FlashImmunityComponent` and then just add a message in code. It's done pretty frequently in the repo and shouldn't be hard to do. There are lots of examples in the codebase.

Helix-ctrl commented:
> you would need to add a subscription for `ExaminedEvent` on `FlashImmunityComponent` and then just add a message in code. It's done pretty frequently in the repo and shouldn't be hard to do. There are lots of examples in the codebase.

I'll give it a try, will update the PR description once i figure it out

Helix-ctrl commented:
> you would need to add a subscription for `ExaminedEvent` on `FlashImmunityComponent` and then just add a message in code. It's done pretty frequently in the repo and shouldn't be hard to do. There are lots of examples in the codebase.

I've been at this for the past 50 minutes and ngl i have no idea how to do this

Samuka-C commented:
![image](https://github.com/user-attachments/assets/526349ee-987e-4a9b-9b3e-a4caa4b9dfd2)

got it working. gonna push the changes here

Samuka-C commented:
I added `SubscribeLocalEvent<FlashImmunityComponent, ExaminedEvent>(OnExamine);` to the Initialize() of the FlashSystem.cs

and added this method (to the FlashSystem.cs):
```
private void OnExamine(EntityUid uid, FlashImmunityComponent component, ExaminedEvent args)
{
       args.PushMarkup(Loc.GetString("flash-protection"));
}
```

and created the `flash-immunity-component.ftl` file in `Resources\Locale\en-US\flash\components\flash-immunity-component.ftl`

and in it I wrote: `flash-protection = It provides protection from bright flashes.`

EmoGarbage404 commented:
meta pr review:

should use the entity<T> version aka
`private void OnExamine(Entity<FlashImmunityComponent> ent, ref ExaminedEvent args)`

also the examine text would be better with color something like
`It provides protection from [color=lightblue]bright flashes[/color].`

Samuka-C commented:
I can't edit the PR, just send suggestions, but thanks for the suggestion to my suggestion :)

Helix-ctrl commented:
> I added `SubscribeLocalEvent<FlashImmunityComponent, ExaminedEvent>(OnExamine);` to the Initialize() of the FlashSystem.cs
> 
> and added this method (to the FlashSystem.cs):
> 
> ```
> private void OnExamine(EntityUid uid, FlashImmunityComponent component, ExaminedEvent args)
> {
>        args.PushMarkup(Loc.GetString("flash-protection"));
> }
> ```
> 
> and created the `flash-immunity-component.ftl` file in `Resources\Locale\en-US\flash\components\flash-immunity-component.ftl`
> 
> and in it I wrote: `flash-protection = It provides protection from bright flashes.`

omfg this is what i was doing i just did it in the wrong file

Helix-ctrl commented:
> Instead of being a description I feel like this should just be automatic examine text that gets applied.

Requested change made. see recent commits 
![image](https://github.com/user-attachments/assets/ea68afb4-ed00-4e7a-9001-4ea773fb5999)

Helix-ctrl commented:
changed PR description and CL to better fit changes

Samuka-C commented on Content.Server/Flash/FlashSystem.cs:
```diff
@@ -54,7 +54,12 @@ public override void Initialize()
         }
-
+        
+        private void OnExamine(EntityUid uid, FlashImmunityComponent component, ExaminedEvent args)
```
Should use the entity version aka
private void OnExamine(Entity<FlashImmunityComponent> ent, ref ExaminedEvent args)

copied EMO suggestion

Helix-ctrl replied:
changed it

Helix-ctrl replied:
change causing tests to fail

Helix-ctrl replied:
fixed issue

Helix-ctrl commented:
all requested changes have been completed afaik

Samuka-C reviewed (approved):
For me is fine, but I am no maint

Samuka-C commented:
> > For me is fine, but I am no maint
> 
> still more experience than me lol, i really appreciate your help 

happy to help

Samuka-C commented:
@EmoGarbage404 can you review it? I think it is fine
